sockeye/inference.py

```diff
@@ -630,6 +648,63 @@ def make_input(sentence_id: int, sentence: str) -> TranslatorInput:
         tokens = list(data_io.get_tokens(sentence))
         return TranslatorInput(id=sentence_id, sentence=sentence.rstrip(), tokens=tokens)
 
+    def batch_translate(self, trans_inputs: List[TranslatorInput]) -> List[TranslatorOutput]:
+        """
+        Batch-translates a list of TranslatorInput, returns a list of TranslatorOutputs
+
+        :param trans_input: List of TranslatorInputs as returned by make_input().
+        :return: List of translation results.
+        """
+        # TODO handle empty inputs, preferably outside inference.py?
+        # TODO avoid accepting and creating lists, use generators?
+
+        # keep input ids to be able to concatenate translated chunks back
+        chunk_markers = []
+        input_chunks = []
+
+        # prepare chunks, keeping track of splitted inputs
+        for tinput in trans_inputs:
+            if len(tinput.tokens) > self.max_input_length:
+                logger.debug("Input %d has length (%d) that exceeds max input length (%d). Splitting into chunks of size %d.",
+                             tinput.id, len(tinput.tokens), self.buckets_source[-1], self.max_input_length)
+                token_chunks = list(utils.chunks(tinput.tokens, self.max_input_length))
+                input_chunks.extend(token_chunks)
+                chunk_markers.extend([tinput.id] * len(token_chunks))
+            else:
+                input_chunks.append(tinput.tokens)
+                chunk_markers.append(tinput.id)
+
+        # start translation in batches
+        translations = []  # type: List[Translation]
+        for i in range(0, len(input_chunks), self.batch_size):
+            logger.debug("Translating batch %d", i)
+            batch = input_chunks[i:i + self.batch_size]
+            # underfilled batch will be filled to a full batch size with empty inputs
+            rest = self.batch_size - len(batch)
+            if rest > 0:
+                logger.debug("Replicating %d random sentences from the last batch to size it to the full batch size (%d)",
+                             rest, self.batch_size)
+                random_indices = np.random.randint(len(batch), size=rest)
```

Comment:
why random? couldn't you pick the same all the time?

Reply:
True, will take the 0th.